pkg/proxy/switch.go

```diff
@@ -215,26 +215,29 @@ func (s *SwitchSession) Bridge(userConn UserConnection, srvConn srvconn.ServerCo
 			nr, err2 := srvConn.Read(buf)
 			validBytes := buf[:nr]
 			if nr > 0 {
-				bufferLen := buffer.Len()
-				if bufferLen > 0 || nr == maxLen {
-					buffer.Write(buf[:nr])
-					validBytes = validBytes[:0]
-				}
-				remainBytes := buffer.Bytes()
-				for len(remainBytes) > 0 {
-					r, size := utf8.DecodeRune(remainBytes)
-					if r == utf8.RuneError {
-						// utf8 max 4 bytes
-						if len(remainBytes) <= 3 {
-							break
+				isZmodem := parser.zmodemParser.IsStartSession()
+				if !isZmodem {
+					bufferLen := buffer.Len()
+					if bufferLen > 0 || nr == maxLen {
+						buffer.Write(buf[:nr])
+						validBytes = validBytes[:0]
+					}
+					remainBytes := buffer.Bytes()
+					for len(remainBytes) > 0 {
+						r, size := utf8.DecodeRune(remainBytes)
+						if r == utf8.RuneError {
+							// utf8 max 4 bytes
+							if len(remainBytes) <= 3 {
+								break
+							}
 						}
+						validBytes = append(validBytes, remainBytes[:size]...)
+						remainBytes = remainBytes[size:]
+					}
+					buffer.Reset()
+					if len(remainBytes) > 0 {
+						buffer.Write(remainBytes)
 					}
-					validBytes = append(validBytes, remainBytes[:size]...)
-					remainBytes = remainBytes[size:]
-				}
-				buffer.Reset()
-				if len(remainBytes) > 0 {
-					buffer.Write(remainBytes)
 				}
 				select {
 				case srvInChan <- validBytes:
```

Comment:
There is no obvious issue with this code snippet. It appears to be a concise implementation of the Bridge() function within a Switch Session class. However, without further context on the use case, I cannot make any optimizations or check for inconsistencies related to knowledge cutoff year 2021.

If you need specific guidance or have questions about its usage, please provide more details about the context in which this function will run and what it's supposed to do within that setting.